app/src/main/java/org/mozilla/fenix/components/toolbar/FenixTabCounterToolbarButton.kt

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.components.browser.state.selector.getNormalOrPrivateTabs
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.toolbar.Toolbar
+import mozilla.components.feature.tabs.toolbar.TabCounterToolbarButton
+import mozilla.components.support.ktx.android.content.res.resolveAttribute
+import mozilla.components.support.utils.DrawableUtils
+import mozilla.components.ui.tabcounter.R
+import mozilla.components.ui.tabcounter.TabCounter
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class FenixTabCounterToolbarButton(
+    val lifecycleOwner: LifecycleOwner,
+    filterPrivate: Boolean? = null,
+    val showTabs: () -> Unit,
+    val store: BrowserStore,
+    val tabCounterMenu: FenixTabCounterMenu
+) : Toolbar.Action,
+    TabCounterToolbarButton(
+        lifecycleOwner,
+        filterPrivate,
+        showTabs,
+        store,
+        tabCounterMenu
+    ) {
+
+    override fun createView(parent: ViewGroup): View {
+        val view = super.createView(parent)
+
+        view.apply {
+            addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
+                override fun onViewAttachedToWindow(v: View?) {
+                    reference.get()?.setCount(
+                        store.state.getNormalOrPrivateTabs(isPrivate(store)).size
+                    )
+                }
+
+                override fun onViewDetachedFromWindow(v: View?) { /* no-op */ }
+            })
+
+            val tint = when (isPrivate(store)) {
+                true -> {
+                    ContextCompat.getColor(context, R.color.primary_text_default_material_dark)
+                }
+                else -> {
+                    ContextCompat.getColor(context, R.color.mozac_ui_tabcounter_default_tint)
+                }
+            }
+            super.reference.get().tintDrawables(tint)
```

Comment:
^^ With this we have the wrong tint on the private browsing home fragment (without a tab). You can reproduce by removing all private tabs, but I think we can actually remove this logic here as well and apply the tint in `HomeFragment.updateTabCounter`.

```kotlin
private fun updateTabCounter(browserState: BrowserState) {
    val tabCount = if (browsingModeManager.mode.isPrivate) {
        view?.tab_button?.tintDrawables(
            ContextCompat.getColor(requireContext(), mozilla.components.ui.tabcounter.R.color.primary_text_default_material_dark)
        )
        browserState.privateTabs.size
    } else {
        view?.tab_button?.tintDrawables(
            ContextCompat.getColor(requireContext(), mozilla.components.ui.tabcounter.R.color.mozac_ui_tabcounter_default_tint)
        )
        browserState.normalTabs.size
    }

    view?.tab_button?.setCountWithAnimation(tabCount)
    view?.add_tabs_to_collections_button?.isVisible = tabCount > 0
}
```

This way we also don't need to make the weak reference accessible (see my comment on the A-C PR: https://github.com/mozilla-mobile/android-components/pull/8872#discussion_r526154128)